src/doc/rustc/src/platform-support/armv7-sony-vita-newlibeabihf.md

```diff
@@ -20,132 +21,62 @@ This target is cross-compiled, and requires installing [VITASDK](https://vitasdk
-dependencies = ["velf"]
-command = "vita-make-fself"
-args = ["-s", '${CARGO_OUT_DIR}/${CARGO_MAKE_CRATE_NAME}.velf', '${CARGO_OUT_DIR}/eboot.bin']
-
-[tasks.param-sfo]
-description = "Build the `param.sfo` manifest using with given TITLE and TITLEID."
-command = "vita-mksfoex"
-args = ["-s", 'TITLE_ID=${TITLEID}', '${TITLE}', '${CARGO_OUT_DIR}/param.sfo']
-
-[tasks.manifest]
-description = "List all static resources into a manifest file."
-script = [
-  'mkdir -p "${CARGO_OUT_DIR}"',
-  '''
-  if [ -d "${STATIC_DIR}" ]; then
-    find "${STATIC_DIR}" -type f > "${CARGO_OUT_DIR}/MANIFEST"
-  else
-    touch "${CARGO_OUT_DIR}/MANIFEST"
-  fi
-  '''
-]
-
-[tasks.vpk]
-description = "Build a VPK distribution of the project executable and resources."
-dependencies = ["eboot-bin", "param-sfo", "manifest"]
-script_runner = "@rust"
-script = [
-    '''
-    use std::io::BufRead;
-    use std::fs::File;
-
-    fn main() {
-
-      let crate_name = env!("CARGO_MAKE_CRATE_NAME");
-      let static_dir = env!("STATIC_DIR");
-      let out_dir = std::path::PathBuf::from(env!("CARGO_OUT_DIR"));
-
-      let mut cmd = ::std::process::Command::new("vita-pack-vpk");
-      cmd.arg("-s").arg(out_dir.join("param.sfo"));
-      cmd.arg("-b").arg(out_dir.join("eboot.bin"));
-
-      // Add files from MANIFEST
-      if let Ok(file) = File::open(out_dir.join("MANIFEST")) {
-          let mut reader = ::std::io::BufReader::new(file);
-          let mut lines = reader.lines();
-          while let Some(Ok(line)) = lines.next() {
-              let p1 = ::std::path::PathBuf::from(line);            // path on FS
-              let p2 = p1.strip_prefix(static_dir).unwrap();        // path in VPK
-              cmd.arg("--add").arg(format!("{}={}", p1.display(), p2.display()));
-          }
-      }
-
-      cmd.arg(out_dir.join(format!("{}.vpk", crate_name)))
-        .output()
-        .expect("command failed.");
-    }
-    '''
-]
+[package.metadata.vita]
+# A unique 9 character alphanumeric identifier of the app.
+title_id = "RUSTAPP01"
+# A title that will be used for the app. Optional, name will be used if not defined
+title_name = "My application"
 ```
 
-After running the above script, you should be able to get a *.vpk file in the same folder your *.elf executable resides. Now you can pick it and install it on your own PlayStation Vita using, or you can use an [Vita3K](https://vita3k.org/) emulator.
+To build vpk with elf in release mode run:
+
+```sh
+cargo vita build vpk --release
+```
+
+After building a *.vpk file it can be uploaded and installed to PlayStation Vita or [Vita3K](https://vita3k.org/) emulator.
 
 ## Testing
 
-Currently there is no support to run the rustc test suite for this target.
+The default Rust test runner is supported, and tests can be compiled to an elf and packed to a *.vpk file using `cargo-vita` tool. Running individual tests is not possible since passing cli arguments to the executable is not supported on Vita.
```

Comment:
I think you might want to open an issue about this, since this sort of thing seems correctable and there's a T-testing now.

```suggestion
The default Rust test runner is supported, and tests can be compiled to an elf and packed to a *.vpk file using `cargo-vita` tool. Filtering tests is not currently supported since passing command-line arguments to the executable is not supported on Vita, so the runner will always execute all tests.
```